posthog/api/test/test_dashboard.py

```diff
@@ -490,3 +490,22 @@ def test_insights_with_no_insight_set(self):
         )
         response = self.client.get(f"/api/projects/{self.team.id}/dashboards/{dashboard.pk}").json()
         self.assertEqual(response["items"][0]["filters"], {"events": [{"id": "$pageview"}], "insight": "TRENDS"})
+
+    def test_retrieve_dashboard_different_project_with_access(self):
```

Comment:
I would consider adding another test case where the dashboard ID exists but the user has no access to that team, to make sure the default 404 is sent (particularly if it comes from a private project in the same org too)